examples/file-uploader/js/app.js

```diff
@@ -0,0 +1,75 @@
+
+const Type = require('union-type');
+const T = require('ramda/src/T')
+    , assoc = require('ramda/src/assoc')
+    , curry  = require('ramda/src/curry')
+    , compose  = require('ramda/src/compose')
+    , map  = require('ramda/src/map')
+    , invoker = require('ramda/src/invoker') 
+;
+const h = require('snabbdom/h');
+
+const uploadList = require('./list');
+const uploader   = require('./uploader');
+  
+
+// action
+
+const listUpdate = (listAction,model) => {
+  const [state, tasks] = uploadList.update(listAction, model.uploads);
+  return [ assoc('uploads', state, model), 
+           tasks.map( map(Action.Route) ) 
+         ];
+}
+
+const Action = Type({
+  Create: [T, T],
+  Route:  [uploadList.Action]
+});
+
+const update = Action.caseOn({
+  Create: (up,files,model) => {
+    return listUpdate( uploadList.Action.Create(up,files), model );
+  },
+
+  Route: listUpdate
+});
+
+
+// model
+
+const init = () => { return { uploads: uploadList.init() }; }
+
+// view
+
+const view = curry( ({url, headers, action$}, model) => {
```

Comment:
That was intended to be in case you need to set headers on the XHR, but it doesn't need to be parameterized at the top level of the app, for same reason `url` can just be a constant.